grpcutil/dial.go

```diff
@@ -28,8 +29,7 @@ func dial(ctx context.Context, address string, creds credentials.TransportCreden
 		defer cancel()
 		conn, err := (&net.Dialer{Cancel: ctx.Done(), Timeout: timeoutDialer}).Dial("tcp", address)
 		if err != nil {
-			log.WithError(err).Warn("Failed to dial grpc connection")
-			return nil, err
+			return nil, fmt.Errorf("Failed to dial grpc connection: %s", err.Error())
```

Comment:
the types on the connections returned from the net package are useful - eg they implement the `Temporary()`, `Timeout()` methods, see https://golang.org/pkg/net/#AddrError.Temporary - would prefer to preserve these and more generally not use fmt.Errorf to wrap errors anywhere (it's bit us already)  - i know and i know you know that go errors suck, sigh, logging may be best but i'm guessing this is being proposed because there wasn't enough context in the logs?